text/0000-service-worker.md

```diff
@@ -0,0 +1,133 @@
+* Start Date: 2018-08-08
+* RFC PR: (leave this empty)
+* Fusion Issue: (leave this empty)
+
+# Summary
+
+Service worker support in Fusion
+
+# Basic example
+
+Most developers will want to use a default service worker implementation (as defined in the SW plugin) without modification. This is how it is used in this case:
+
+Note: Fusion CLI is hardcoded to look for a `src/sw-logic.js` file (akin to how it uses `src/main.js`).
+```js
+// src/sw-logic.js
+
+import {serviceWorkerLogic} from "fusion-plugin-sw";
+
+export default serviceWorkerLogic;
+```
+
+Add registration code to app:
+```js
+// src/main.js
+import {registerServiceWorker, SWPlugin} from "fusion-plugin-sw";
+
+if (__BROWSER__ && !__DEV__) {
+  registerServiceWorker();
+}
+
+export default () => {
+  const app = new App(/* ... */);
+  app.register(SWPlugin);
+  // ...
+  return app;
+}
+
+```
+
+# Motivation
+
+<!--
+Why are we doing this? What use cases does it support? What is the expected
+outcome?
+
+Please focus on explaining the motivation so that if this RFC is not accepted,
+the motivation could be used to develop alternative solutions. In other words,
+enumerate the constraints you are trying to solve without coupling them too
+closely to the solution you have in mind.
+-->
+
+# Detailed design
+
+Two low-level APIs:
+
+1) New entry file: `src/sw-logic.js`:
+```js
+// src/sw-logic.js
+
+export default ({criticalScripts, staticAssets}) => {
+  // `criticalScripts` - an array of string paths that should be pre-cached
```

Comment:
I'd call them `criticalAssets` and `cacheableAssets`. Potentially, non-scripts could be pre-cached and cacheable assets are often not static assets (e.g. async script chunks)